### PR TITLE
charts/k8s-stack(hack): fix sync_rules.py conditions and replacements 

### PR DESCRIPTION
**Pull request by antoinedeschenes**
Fixes to hack/sync_rules.py:
* Replaces the hardcoded "monitoring" namespace in Alertmanager alerting rules with the release namespace, to match the VMAlertmanager CRD namespace.
* Fixes `kube-apiserver-histogram.rules` and `alertmanager.rules` enabling conditions

### CODE REVIEW

**Comment by f41gh7** [comment]
Hello, these rules downloaded from https://raw.githubusercontent.com/prometheus-operator/kube-prometheus/main/manifests/alertmanager-prometheusRule.yaml.

 I think, this line must be updated somehow

https://github.com/VictoriaMetrics/helm-charts/blob/master/charts/victoria-metrics-k8s-stack/hack/sync_rules.py#L136

cc @k1rk

File: charts/victoria-metrics-k8s-stack/hack/sync_rules.py [hunk 1]
```diff
@@ -98,7 +99,7 @@ def new_representer(dumper, data):
     'node-network': '.Values.defaultRules.rules.network',
     'node.rules': '.Values.defaultRules.rules.node',
     'vmagent': '.Values.vmagent.enabled .Values.defaultRules.rules.vmagent',
-    'alertmaanger': '.Values.defaultRules.rules.alertmanager',
```
Comment (antoinedeschenes) [inline charts/victoria-metrics-k8s-stack/hack/sync_rules.py:101]: Must match the [alertgroup name](https://github.com/VictoriaMetrics/helm-charts/pull/537/files#diff-be7fe52ad55b3eee294f8f15d6bd822251f33b40f76d35f45a4a837447f4c562L24) to properly set the condition.

File: charts/victoria-metrics-k8s-stack/templates/victoria-metrics-operator/_rules.tpl
Comment (antoinedeschenes) [inline charts/victoria-metrics-k8s-stack/templates/victoria-metrics-operator/_rules.tpl:1]: I haven't found that `define` template used anywhere

File: charts/victoria-metrics-k8s-stack/templates/rules/kube-apiserver-histogram.rules.yaml
Comment (antoinedeschenes) [inline charts/victoria-metrics-k8s-stack/templates/rules/kube-apiserver-histogram.rules.yaml:1]: Changes from running sync_rules.py.

File: charts/victoria-metrics-k8s-stack/templates/rules/alertmanager.rules.yaml
Comment (antoinedeschenes) [inline charts/victoria-metrics-k8s-stack/templates/rules/alertmanager.rules.yaml:1]: Changes from running sync_rules.py.

File: charts/victoria-metrics-k8s-stack/hack/sync_rules.py [hunk 2]
```diff
@@ -76,8 +76,9 @@ def new_representer(dumper, data):
     'etcd': '.Values.kubeEtcd.enabled .Values.defaultRules.rules.etcd',
     'general.rules': '.Values.defaultRules.rules.general',
     'k8s.rules': '.Values.defaultRules.rules.k8s',
-    'kube-apiserver-burnrate.rules': '.Values.kubeApiServer.enabled .Values.defaultRules.rules.kubeApiserverBurnrate',
     'kube-apiserver-availability.rules': '.Values.kubeApiServer.enabled .Values.defaultRules.rules.kubeApiserverAvailability',
+    'kube-apiserver-burnrate.rules': '.Values.kubeApiServer.enabled .Values.defaultRules.rules.kubeApiserverBurnrate',
+    'kube-apiserver-histogram.rules': '.Values.kubeApiServer.enabled .Values.defaultRules.rules.kubeApiserverHistogram',
```
Comment (antoinedeschenes) [inline charts/victoria-metrics-k8s-stack/hack/sync_rules.py:81]: Condition was missing for the matching VMRule

**Comment by antoinedeschenes** [comment]
@f41gh7 Updated sync_rules.py to generate the changes. Also fixed 2 rule conditions.

**Review by f41gh7 — APPROVED** [review]
LGTM

**Comment by f41gh7** [comment]
Thanks for contribution!

**Pull request merged**